flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperatorV2.java

```diff
@@ -268,7 +296,11 @@ protected boolean isUsingCustomRawKeyedState() {
      * @throws Exception An exception in this method causes the operator to fail.
      */
     @Override
-    public void open() throws Exception {}
+    public void open() throws Exception {
+        this.watermarkOutputMultiplexer =
+                setupWatermarkMultiplexer(
+                        containingTask.getStreamStatusMaintainer(), numberOfInputs);
+    }
```

Comment:
Why are you initialising this non final field in the open method? It probably should happen in the constructor and `watermarkOutputMultiplexer` should be final?

Reply:
Fully agree, the problem is that the `containingTask.getStreamStatusMaintainer()` is not set until the `open` method.